src/tools/ceph-dencoder/rbd_types.h

```diff
@@ -49,4 +54,22 @@ TYPE(cls::rbd::GroupSpec)
 TYPE(cls::rbd::ImageSnapshotSpec)
 TYPE(cls::rbd::SnapshotInfo)
 TYPE(cls::rbd::SnapshotNamespace)
+using namespace cls::rbd;
+TYPE(ParentImageSpec)
+TYPE(ChildImageSpec)
+TYPE(MigrationSpec)
+TYPE(MirrorPeer)
+TYPE(MirrorImage)
+TYPE(MirrorImageMap)
+TYPE(MirrorImageStatus)
+TYPE(MirrorImageSiteStatus)
+TYPE_FEATUREFUL(MirrorImageSiteStatusOnDisk)
+TYPE(GroupImageSpec)
+TYPE(GroupImageStatus)
+TYPE(GroupSnapshot)
+TYPE(GroupSpec)
+TYPE(ImageSnapshotSpec)
+TYPE(SnapshotInfo)
+TYPE(SnapshotNamespace)
```

Comment:
We need to decide which way to go for types which are not matching or missing: introduce them either as fully qualified or not fully qualified and be consistent about it.  This is actually an interesting question because I don't think we have ever intended for C++ namespaces to "leak" in this fashion.  Regardless, I really doubt that both
```
TYPE(cls::rbd::SnapshotNamespace)
```
and
```
TYPE(SnapshotNamespace)
```
are needed and this goes for 15 other types above too.

The only type which _has_ to be fully qualified is `NotifyMessage` because there are five of them in different namespaces.